pkg/subvolume/subvolume.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2023 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package subvolume
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	exec "github.com/rook/kubectl-rook-ceph/pkg/exec"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	"github.com/rook/kubectl-rook-ceph/pkg/logging"
+)
+
+func List(ctx context.Context, clientsets *k8sutil.Clientsets, clusterNamespace, operatorNamespace string) {
+
+	getpv := `kubectl get pv -o jsonpath={.items[*].spec.csi.volumeAttributes.subvolumeName}`
+	// Get all csi reference on k8s side for all cephfs pv
+	k8sRef := exec.ExecuteBashCommand(getpv)
+	if k8sRef == "" {
+		logging.Fatal(fmt.Errorf("No k8s pvs found. Nothing to do."))
+	}
+	k8sRefl := strings.Split(k8sRef, " ")
+	//logging.Info("k8s Ref found:%s", k8sRefl)
+
+	// get the csi reference on CSI side for all cephfs PV
+	svCSIRef := getCSIRef(ctx, clientsets, clusterNamespace, operatorNamespace)
+
+	// Check whether difference is of the two lists is measurable
+	diff := getDifference(svCSIRef, k8sRefl)
+	if diff == nil {
+		logging.Info("No stale volumes found")
+	}
+	fmt.Println("list of stale volumes to be deleted", diff)
+}
+
+func getDifference(csiref, k8ref []string) []string {
+	mb := make(map[string]struct{}, len(k8ref))
+	for _, x := range k8ref {
+		mb[x] = struct{}{}
+	}
+	var diff []string
+	for _, x := range csiref {
+		if _, found := mb[x]; !found {
+			// reference present in ceph but not in k8s
+			diff = append(diff, x)
+		}
+	}
+	if diff == nil {
+		logging.Fatal(fmt.Errorf("No stale Volume found"))
+	}
+
+	return diff
+}
+
+func getCSIRef(ctx context.Context, clientsets *k8sutil.Clientsets, clusterNamespace, operatorNamespace string) []string {
+
+	fsList := exec.RunCommandInOperatorPod(ctx, clientsets, "ceph", []string{"fs", "ls", "--format", "json"}, clusterNamespace, operatorNamespace, true, false)
+	if fsList == "" {
+		logging.Fatal(fmt.Errorf("failed to get filesystem"))
+	}
+	type fsStruct struct {
+		Name string
+		data []string
+	}
+	var fsstruct []fsStruct
+	var subvolg []fsStruct
+	var subvol []fsStruct
```

Comment:
```suggestion
	var fsstruct []fsStruct
	var subvolg []fsStruct
	var subvol []fsStruct
```

declare the var where used so that it is easier to read